app/src/components/data-grid/autocomplete/AsyncAutocompleteDataGridEditCell.tsx

```diff
@@ -200,11 +203,14 @@ const AsyncAutocompleteDataGridEditCell = <DataGridType extends GridValidRowMode
         />
       )}
       renderOption={(renderProps, renderOption) => {
-        return (
-          <Box component="li" {...renderProps}>
-            {renderOption.label}
-          </Box>
-        );
+        if (!props.renderOption) {
+          return (
+            <Box component="li" {...renderProps}>
```

Comment:
In reference to my prev comment, the use of `renderProps` is ambiguous because the prop is being passed to two different components in these two branches.

Since Autocomplete options always have to be list items (unless I'm mistaken), we should probably move `props.renderOption` inside the `<li>`. Then, if we want to customize the `<li>`, we should add a `ListItemProps?: BoxProps`
```javascript
return (
  <Box component='li' {...props.ListItemProps}>
    {props.renderOption ? props.renderOption(option) : <span>{option.label}</span>}
  </Box>
)
```